mongodb-plugin/src/main/java/io/cdap/plugin/batch/sink/MongoDBBatchSink.java

```diff
@@ -56,42 +57,56 @@
 public class MongoDBBatchSink extends ReferenceBatchSink<StructuredRecord, NullWritable, BSONWritable> {
 
   private final MongoDBConfig config;
+  private RecordToBSONWritableTransformer transformer;
 
   public MongoDBBatchSink(MongoDBConfig config) {
     super(new ReferencePluginConfig(config.referenceName));
     this.config = config;
   }
 
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    config.validate();
```

Comment:
should we have input schema validation?

Reply:
[MongoDBBatchSink#validateInputSchema method created](https://github.com/vladhlinskiy/database-plugins/pull/14/files#diff-6442105b7c58b2a7fd70e331659527bcR109) to check for supported types:

```
  private void validateInputSchema(Schema inputSchema) {
    List<Schema.Field> fields = inputSchema.getFields();
    if (fields == null || fields.isEmpty()) {
      throw new InvalidStageException("Input schema should contain fields");
    }
    for (Schema.Field field : fields) {
      Schema.Type fieldType = field.getSchema().isNullable() ?
        field.getSchema().getNonNullable().getType() : field.getSchema().getType();
      if (!SUPPORTED_FIELD_TYPES.contains(fieldType)) {
        String supportedTypes = SUPPORTED_FIELD_TYPES.stream()
          .map(Enum::name)
          .map(String::toLowerCase)
          .collect(Collectors.joining(", "));
        String errorMessage = String.format("Field '%s' is of unsupported type '%s'. Supported types are: %s.",
                                            field.getName(), fieldType, supportedTypes);
        throw new InvalidStageException(errorMessage);
      }
    }
  }
```